clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerProtocolTest.java

```diff
@@ -247,4 +245,42 @@ public void deserializeFutureAssignmentVersion() {
         Assignment assignment = ConsumerProtocol.deserializeAssignment(buffer);
         assertEquals(toSet(Collections.singletonList(tp1)), toSet(assignment.partitions()));
     }
+
+    private ByteBuffer generateFutureSubscriptionVersionData(boolean hasGenerationId) {
+        // verify that a new version which adds a field is still parseable
+        short version = 100;
+
+        Schema subscriptionSchemaV100 = new Schema(
+            new Field("topics", new ArrayOf(Type.STRING)),
+            new Field("user_data", Type.NULLABLE_BYTES),
+            new Field("owned_partitions", new ArrayOf(
+                ConsumerProtocolSubscription.TopicPartition.SCHEMA_1)),
+            hasGenerationId ? new Field("generation_id", Type.INT32) : new Field("foo", Type.STRING),
```

Comment:
This looks weird to me. If we are testing with version 100, `generation_id` should always be there, no? The latest version of the subscription has it. `ConsumerProtocol.deserializeAssignment()` will actually use the latest version, which includes generation id, to decode the payload here regardless of this. If you don't put it, the generation may get intitialized based on the `bar`'s data.

Reply:
I think you're right. The generation_id should always be there. I think this test is just to test we will always get the correct data no matter what new fields added. Updated the test.